### PR TITLE
Setup jest

### PR DESCRIPTION
**Pull request by kapke**

### CODE REVIEW

File: Deceiver.ts [hunk 1]
```diff
@@ -22,27 +21,29 @@ export class DeceiverMirror<T, K extends keyof T> {
...
             .map(prototype => ({prototype, names: Object.getOwnPropertyNames(prototype)}))
             .map(({prototype, names}) => ({
                 prototype,
                 names: names.filter(name => Object.getOwnPropertyDescriptor(prototype, name).get),
             }))
             .flatMap(({names}) => names)
             .filter(name => name !== '__proto__')
-            .reduce(this.toUniqueArray, []);
+            .reduce(this.toUniqueArray, [])
+            .value();
```
Comment (cparjaszewski) [inline Deceiver.ts]: These methods: `getMethodNames` and `getPropertyNames` - look very similar to each other, is there any way to abstract them?

I think the only unique lines in both are these:
```.filter(name => name !== '__proto__')``` vs ```.filter(name => name != 'constructor')```
```names: names.filter(name => typeof Object.getOwnPropertyDescriptor(prototype, name).value == 'function'),``` vs  ```names: names.filter(name => Object.getOwnPropertyDescriptor(prototype, name).get),
              }))		              }))```

File: Deceiver.ts [hunk 2]
```diff
@@ -22,27 +21,29 @@ export class DeceiverMirror<T, K extends keyof T> {
...
     public getMethodNames (): K[] {
-        return _(this.getAllPrototypes())
+        return _.chain(this.getAllPrototypes())
             .map(prototype => ({prototype, names: Object.getOwnPropertyNames(prototype)}))
             .map(({prototype, names}) => ({
                 prototype,
                 names: names.filter(name => typeof Object.getOwnPropertyDescriptor(prototype, name).value == 'function'),
             }))
             .flatMap(({names}) => names)
             .filter(name => name != 'constructor')
```
Comment (tomaszferens) [inline Deceiver.ts]: maybe you can use `!==` instead.

Comment (kapke) [reply Deceiver.ts]: It's already changed in PR with prettier to have `!=` in both `getMethodNames` and `getPropertyNames`

File: Deceiver.ts [hunk 3]
```diff
@@ -22,27 +21,29 @@ export class DeceiverMirror<T, K extends keyof T> {
...
     public getPropertyNames (): K[] {
-        return _(this.getAllPrototypes())
+        return _.chain(this.getAllPrototypes())
             .map(prototype => ({prototype, names: Object.getOwnPropertyNames(prototype)}))
             .map(({prototype, names}) => ({
                 prototype,
                 names: names.filter(name => Object.getOwnPropertyDescriptor(prototype, name).get),
             }))
             .flatMap(({names}) => names)
             .filter(name => name !== '__proto__')
```
Comment (kamilpodlasek) [inline Deceiver.ts]: You should use `!=` to be consistent

**Review by kamilpodlasek — APPROVED** [review]

File: Deceiver.ts (hunk 1 above)
Comment (kapke) [reply Deceiver.ts]: I will check how this code could look like on branch with memoization

File: Deceiver.ts (hunk 3 above)
Comment (kapke) [reply Deceiver.ts]: Done on branch with prettier

**Pull request merged**